circuit_maintenance_parser/parsers/cogent.py

```diff
@@ -28,50 +28,64 @@ def parse_html(self, soup, data_base):
         """Execute parsing."""
         data = data_base.copy()
         try:
-            self.parse_br(soup.find_all("br"), data)
+            self.parse_div(soup.find_all("div", class_="a3s aiL"), data)
             self.parse_title(soup.find_all("title"), data)
             return [data]
 
         except Exception as exc:
             raise ParsingError from exc
 
-    def parse_br(self, br_results: ResultSet, data: Dict):
-        """Parse <br> tag."""
-        for br in br_results:
-            div_text = br.parent.text
-            for line in re.split(r"\n", div_text):
+    def parse_div(self, divs: ResultSet, data: Dict):  # pylint: disable=too-many-locals
+        """Parse <div> tag."""
+        start_str = ""
+        end_str = ""
+
+        for div in divs:
+            for line in div.text.splitlines():
                 if line.endswith("Network Maintenance"):
                     data["summary"] = line
                 elif line.startswith("Dear"):
-                    match = re.search("Dear (.*),", line)
+                    match = re.search(r"Dear (.*),", line)
                     if match:
                         data["account"] = match.group(1)
                 elif line.startswith("Start time:"):
-                    match = re.search("Start time: (.*) \(local time\) (\d+/\d+/\d+)", line)
+                    match = re.search(r"Start time: (.*) \([A-Za-z\s]+\) (\d+/\d+/\d+)", line)
                     if match:
                         start_str = " ".join(match.groups())
-                        start = datetime.strptime(start_str, "%I:%M %p %d/%m/%Y")
-                        data["start"] = self.dt2ts(start)
                 elif line.startswith("End time:"):
-                    match = re.search("End time: (.*) \(local time\) (\d+/\d+/\d+)", line)
+                    match = re.search(r"End time: (.*) \([A-Za-z\s]+\) (\d+/\d+/\d+)", line)
                     if match:
                         end_str = " ".join(match.groups())
+                elif line.startswith("Cogent customers receiving service"):
+                    match = re.search(r"[^Cogent].*?((\b[A-Z][a-z\s-]+)+, ([A-Za-z-]+[\s-]))", line)
+                    if match:
+                        parsed_timezone = city_timezone(match.group(1).strip())
+                        local_timezone = timezone(parsed_timezone)
+                        # set start time using the local city timezone
+                        start = datetime.strptime(start_str, "%I:%M %p %d/%m/%Y")
+                        local_time = local_timezone.localize(start)
+                        # set start time to UTC
+                        utc_start = local_time.astimezone(UTC)
+                        data["start"] = self.dt2ts(utc_start)
```

Comment:
Nice! For troubleshooting purposes, it would probably be good to include a log message or two here and below, maybe something along the lines of:

```python
logger.info("Mapped %s at %s (%s), to %s (UTC)", start_str, match.group(1), local_timezone, utc_start)
```

Reply:
Good idea Glenn. Start and End logging added:
`INFO     circuit_maintenance_parser.parsers.cogent:cogent.py:70 Mapped start time 12:00 AM 06/11/2021 at San Jose, CA (America/Los_Angeles), to 2021-11-06 07:00:00+00:00 (UTC)`
`INFO     circuit_maintenance_parser.parsers.cogent:cogent.py:83 Mapped end time 03:00 AM 06/11/2021 at San Jose, CA (America/Los_Angeles), to 2021-11-06 10:00:00+00:00 (UTC)`